### PR TITLE
deprecate 'priv' argument in postgresql_user

### LINKED ISSUE #212: remove priv argument to the postgresql_user module
##### SUMMARY

Given that the `postgresql_privs` module exists for granting/revoking permissions on relations to roles, the `priv` argument of the `postgresql_user` module seems redundant.  I understand that it is probably convenient for folks, but as I see it:
- its duplicate functionality
- its duplicate code
- it muddies two different activities into one module (which seems anti-KISS)

As such, I propose deprecating and then (eventually) removing it.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
postgresql_user

### PR DESCRIPTION
**Pull request by hunleyd**
##### SUMMARY
Deprecate the `priv` argument of postgresql_user in favor of the postgresql_privs module

fixes #212

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
postgresql_user

### CODE REVIEW

**Comment by hunleyd** [comment]
> Unsupported parameters for (ansible.legacy.command) module: warn

File: plugins/modules/postgresql_user.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 - Set I(fail_on_user) to C(no) to make the module ignore failures when trying to remove a user.
   In this case, the module reports if changes happened as usual and separately reports
   whether the user has been removed or not.
+- B(WARNING) The C(priv) option has been B(deprecated) and will be removed in community.postgresql 3.0.0. Please use the I(postgresql_privs) module instead.
```
Comment (Andersson007) [inline plugins/modules/postgresql_user.py:24]:
```suggestion
- B(WARNING) The I(priv) option has been B(deprecated) and will be removed in community.postgresql 3.0.0. Please use the M(community.postgresql.postgresql_privs) module instead.
```
this is a requirement from the official doc, i.e. modules have to use M(FQCN)

File: plugins/modules/postgresql_user.py [hunk 2]
```diff
@@ -58,6 +59,9 @@
     - fail_on_role
   priv:
     description:
+    - This option has been B(deprecated) and will be removed in
+      community.postgresql 3.0.0. Please use the I(postgresql_privs) module to
```
Comment (Andersson007) [inline plugins/modules/postgresql_user.py:63]:
```suggestion
      community.postgresql 3.0.0. Please use the M(community.postgresql.postgresql_privs) module to
```

**Comment by Andersson007** [comment]
> > Unsupported parameters for (ansible.legacy.command) module: warn
> 
> that doesn't appear to be my error. is the pipeline ok @Andersson007 ?

@hunleyd Thanks for the PR!
Please take a look at `tests/integration/targets/setup_postgresql_db/tasks/main.yml
```
163   - name: Reinstall internationalization files
164     shell: yum -y reinstall glibc-common || yum -y install glibc-common
165     args:
166       warn: false
167     when: locale_present is failed
```
I think some args have been deprecated here, I would re-write it somehow

**Comment by Andersson007** [comment]
@hunleyd As it's not something directly related, feel free to submit a separate PR. It would be much easier to backport. Once it's merged, we can re-launch the tests here.

**Comment by hunleyd** [comment]
> I think some args have been deprecated here (or something else happened related to the `warn:` thing), I would re-write it somehow. I would start with removing `args: warn: false`.

According to the [docs](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/shell_module.html) that arg is still valid though which is why I'm confused.

**Comment by Andersson007** [comment]
I'm also confused because https://github.com/ansible-collections/community.postgresql/pull/228 has passed...

**Comment by Andersson007** [comment]
Ah, no, it was just skipped

**Comment by Andersson007** [comment]
@hunleyd it's not valid https://docs.ansible.com/ansible/devel/collections/ansible/builtin/shell_module.html. We should choose `devel` in the upper left corner in this case

**Comment by Andersson007** [comment]
One of changelog fragments in ansible/ansible:
```
  1 removed_features:
  2   - Remove deprecated ``COMMAND_WARNINGS`` configuration option (https://github.com/ansible/ansible/issues/77394)
  3   - command/shell - remove deprecated ``warn`` module param
```

**Comment by hunleyd** [comment]
`Error when getting collection version metadata for community.general:3.0.2 from default (https://galaxy.ansible.com/api/) (HTTP Code: 429, Message: Too Many Requests Code: Unknown)`
guess i'll try again later ;)

**Comment by Andersson007** [comment]
> `Error when getting collection version metadata for community.general:3.0.2 from default (https://galaxy.ansible.com/api/) (HTTP Code: 429, Message: Too Many Requests Code: Unknown)` guess i'll try again later ;)

yep, we should adjust things to install `c.general` from git instead (i always forget about it)

**Comment by Andersson007** [comment]
Could you please take a look at https://github.com/ansible-collections/community.postgresql/pull/227#discussion_r841185678 and  https://github.com/ansible-collections/community.postgresql/pull/227#discussion_r841185715 ?

**Pull request merged**